tests/TestCase/Authenticator/FormAuthenticatorTest.php

```diff
@@ -236,6 +268,38 @@ public function testMultipleLoginUrlSuccess()
         $this->assertEquals([], $result->getErrors());
     }
 
+    /**
+     * testLoginUrlSuccessWithBase
+     *
+     * @return void
+     */
+    public function testLoginUrlSuccessWithBase()
+    {
+        $identifiers = new IdentifierCollection([
+            'Authentication.Password',
+        ]);
+
+        $request = ServerRequestFactory::fromGlobals(
+            ['REQUEST_URI' => '/users/login'],
+            [],
+            ['username' => 'mariano', 'password' => 'password']
+        );
+        $uri = $request->getUri();
+        $uri->base = '/base';
+        $request = $request->withUri($uri);
+        $response = new Response();
+
+        $form = new FormAuthenticator($identifiers, [
+            'loginUrl' => '/base/users/login',
+        ]);
+
+        $result = $form->authenticate($request, $response);
+
+        $this->assertInstanceOf(Result::class, $result);
+        $this->assertEquals(Result::SUCCESS, $result->getStatus());
```

Comment:
assertSame() should ideally be used for scalar comparison to avoid false positive on casting.

Reply:
Lazy me just copied the tests :) But aren't both values supposed to be strings?

Reply:
Could be, sure.